ubirch/ubirch_protocol.go

```diff
@@ -19,21 +19,24 @@
 package ubirch
 
 import (
+	"crypto/sha256"
 	"errors"
 	"fmt"
 
 	"github.com/google/uuid"
 	"github.com/ugorji/go/codec"
 )
 
+// ProtocolType definition
 type ProtocolType uint8
 
 const (
-	Plain   ProtocolType = 0x00
-	Signed  ProtocolType = 0x22
-	Chained ProtocolType = 0x23
+	Plain   ProtocolType = 0x00 // Plain protocol, without hashing and signing
```

Comment:
This seems inconsistent across implementations, see e.g. here: https://github.com/ubirch/ubirch-protocol/blob/7a2b43be54745cba24e974a35be42a13b5f32b1a/ubirch/ubirch_protocol.h#L82 Plain type is 0x00 here and 0x01 in the c implementation. This needs to be 0x21.

Reply:
fixed